continuous-deployment/secrets-management/readme.md

```diff
@@ -1,129 +1,122 @@
+
+> If you are working on an application with enhanced security requirements you should consider using additional techniques to maintain encryption on secrets throughout the application lifetime.
+
+Always rotate encryption keys on a regular basis.
+
+## Techniques for Secrets Management
+
+These techniques make the loading of secrets  transparent to the developer.
+
+### C#/.NET
+
+Use the [`file`](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/appsettings/appsettings-element-for-configuration) attribute of the appSettings element to load secrets from a local file.
 
 ``` XML
 <?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <appSettings file="..\..\secrets.config">
-  ...
+  …
   </appSettings>
   <startup>
       <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6.1" />
   </startup>
-  ...
+  …
 </configuration>
 ```
 
-Secrets can then be accessed like so:
+Access secrets:
 
 ```C#
-
-    static void Main(string[] args)
-    {
-      String mySecret = System.Configuration.ConfigurationManager.AppSettings["mySecret"];
-    }
+static void Main(string[] args)
+{
+    String mySecret = System.Configuration.ConfigurationManager.AppSettings["mySecret"];
+}
 ```
 
-When running in Azure ConfigurationManager will load these settings from the process environment. No secrets file need to be uploaded to the server and no code needs to be changed.
+When running in Azure, ConfigurationManager will load these settings from the process environment. We don't need to upload secrets files to the server or change any code.
 
-#### Node
+### Node
 
-Use the [dotenv](https://www.npmjs.com/package/dotenv) package.
+Store secrets in environment variables or in a `.env` file
 
 ```bash
 $ cat .env
 MY_SECRET=mySecret
 ```
 
+Use the [dotenv](https://www.npmjs.com/package/dotenv) package to load and access environment variables
+
 ```node
 require('dotenv').config()
 let mySecret = process.env("MY_SECRET")
 ```
 
-Secrets are loaded into the environment from the .env file if it exists. If not just use existing environment i.e. ApplicationSettings.
-
-#### Python
+### Python
 
-Load config settings into environment manually.
+Store secrets in environment variables or in a `.env` file
 
 ```bash
-$ cat secrets.json
-{
-  "MY_SECRET" : "mySecret"
-}
+$ cat .env
+MY_SECRET=mySecret
 ```
 
-```Python
+Use the [dotenv](https://pypi.org/project/python-dotenv/) package to load and access environment variables
 
+```Python
 import os
-import json
+from dotenv import load_dotenv
 
-try:
-  config = json.loads(open("./secrets.json", "r").read())
-
-  # Works for all versions
-  for k,v in config.items():
-    os.environ[k] = v
-
-  # Python >= 3.5 one-liner
-  # os.environ = { **os.environ, **config }
-
-except IOError:
-  pass # No secrets.json
-except json.decoder.JSONDecodeError:
-  pass # Malformed secrets.json
-
-print(os.environ["MY_SECRET"])
 
+load_dotenv()
+my_secret = os.getenv('MY_SECRET')
 ```
```

Comment:
I would also add the environs library as its another great one! 

```
from environs import Env
env = Env()
env.read_env()
my_secret = os.environ["MY_SECRET"]
```

Reply:
added